pkg/virt-launcher/virtwrap/util/libvirt_helper.go

```diff
@@ -414,5 +415,12 @@ func SetupLibvirt() error {
 		}
 	}
 
-	return nil
+	return err // Need to use err and not nil to allow the defered func to set the value if needed
+}
+
+func fileClose(c io.Closer, err *error) {
```

Comment:
I think you can generalize this by passing in a close function (`func() error`)

Reply:
Not sure I follow - can you elaborate what you propose ?

Reply:
Sorry, the func name confused me. This is a generic close, not file specific. Maybe call it CloseWithErr or something similar.